src/components/views/rooms/RoomPreviewBar.js

```diff
@@ -106,12 +108,24 @@ module.exports = React.createClass({
     },
 
     _checkInvitedEmail: async function() {
-        // If this is an invite and we've been told what email
-        // address was invited, fetch the user's list of Threepids
-        // so we can check them against the one that was invited
+        // If this is an invite and we've been told what email address was
+        // invited, fetch the user's account emails and discovery bindings so we
+        // can check them against the email that was invited.
         if (this.props.inviterName && this.props.invitedEmail) {
             this.setState({busy: true});
             try {
+                // Gather the account 3PIDs
+                const account3pids = await MatrixClientPeg.get().getThreePids();
+                this.setState({
+                    accountEmails: account3pids.threepids.filter(b => b.medium === 'email')
+                                                         .map(b => b.address),
```

Comment:
4 space indent maximum according to our style choices (or just stick it on the end of the above line - it's surely within the length limit?)

Reply:
Ah yes, I guess we don't really use this style here. 😅 I'll tweak it.